graphql/project_resolver.go

```diff
@@ -42,7 +42,7 @@ func (r *projectResolver) Patches(ctx context.Context, obj *restModel.APIProject
 	apiPatches := []*restModel.APIPatch{}
 	for _, p := range patches {
 		apiPatch := restModel.APIPatch{}
-		err = apiPatch.BuildFromService(p)
+		err = apiPatch.BuildFromService(p, &restModel.APIPatchArgs{IncludeProjectIdentifier: true})
```

Comment:
I think we would need to include the [childPatches](https://github.com/evergreen-ci/evergreen/blob/main/graphql/schema/types/patch.graphql#L64) in the two graphql resolvers (and tests) as well.

Reply:
Yup thats correct! We use it on the version page, and on the configure page but we don't use it on the general "My Patches" page.

For ProjectIdentifier i think a hybrid solution would work best. Like we do for the taskResolver projectIdentifier.
https://github.com/evergreen-ci/evergreen/blob/34ccd3df43c009ba56a7fcf657a6c2b8516f0716/graphql/task_resolver.go#L455-L458

A solution where we would have a fallback to the  other source would be ideal. 

Depending on how the patch model is built the projectIdentifier could be included from the `BuildFromService` but if not having the fallback to its own resolver would ensure that we never accidentally omit this field without introducing the risk of doing an unnecessary additional query.

Reply:
What did we end up deciding with regards to child patches? It's currently being fetched on the main patch query so I think we need to add it in somehow? The ideal set up would be to put it in its own resolver, but at the very least we need it in the main query (and then maybe move it as an optimization in a separate ticket).

Reply:
@malikchaya2 the actual FindPatchById that is used by the Patch resolver _does_ fetch it since it uses FindPatchById which loads everything so your request is already handled. https://github.com/evergreen-ci/evergreen/blob/25f010ae0c90c76ddcb40b036735a8ee9fabcb16/rest/data/patch.go#L82

if we do want a separate resolver for it, that'll be more complicated than I know how to do since the child patches function modifies many variables, not just one, so UI could take that optimization ticket if it seems worth doing. cc @khelif96

Reply:
Looking at the [project patches](https://github.com/evergreen-ci/spruce/blob/main/src/gql/queries/project-patches.graphql#L1), the child patches are actually not being fetched there, so that makes sense to me.

Reply:
Ah that was also discussed above 👍 
> We use it on the version page, and on the configure page but we don't use it on the general "My Patches" page.